.github/workflows/rule-validate.yml

```diff
@@ -36,43 +36,56 @@ jobs:
         id: comment_branch
 
       - name: Get Refs
-        id: get_refs
+        id: get_head_ref
         run: |
+          # Accurate for push events, other events require changes.
           head_ref="${{ github.head_ref }}"
           repo="${{ github.repository }}"
-          # Either of these must be set in block below
-          run_all=""
-          base_ref=""
 
           if [[ "${{ github.event_name }}" == 'pull_request_target' ]]; then
-            # Detect changes based on whatever we're merging into.
-            base_ref="${{ github.base_ref }}"
             repo="${{ github.event.pull_request.head.repo.full_name }}"
-          elif [[ "${{ github.event_name }}" == 'push' ]]; then
-            # Detect changes based on the previous commit
-            base_ref="$(git rev-parse HEAD^)"
```

Comment:
This can't be done until after the Checkout step, so I had to split this into two steps.